### PR TITLE
feat(messaging): add support for exporting delivery metrics to BigQuery

### LINKED ISSUE #8538: Add steps to enable delivery report exports to big query.
Currently there is firebase documentation for adding message delivery exports to biq query only for the Android, iOS, and Web platforms.

The FlutterFire docs should have an article how to do this in Flutter.

https://firebase.google.com/docs/cloud-messaging/understand-delivery?platform=ios#enable-message-delivery-data-export

### PR DESCRIPTION
**Pull request by Lyokone**
## Description

In order to support deliveryMetricsExportedToBigQueryEnabled on the Web, I had to set up a service worker with a bundler
The reasons:
- The v9 compat version does not includes the needed deliveryMetricsExportedToBigQueryEnabled method
- By setting the service worker type to module, the Firebase Messaging service worker fails because it does not support es modules

## Related Issues

closes #8538

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_messaging/firebase_messaging_platform_interface/lib/src/platform_interface/platform_interface_messaging.dart [hunk 1]
```diff
@@ -305,4 +305,11 @@ abstract class FirebaseMessagingPlatform extends PlatformInterface {
   Future<void> unsubscribeFromTopic(String topic) {
     throw UnimplementedError('unsubscribeFromTopic() is not implemented');
   }
+
+  /// Enables or disables Firebase Cloud Messaging message delivery metrics export to BigQuery.
```
Comment (russellwheatley) [inline packages/firebase_messaging/firebase_messaging_platform_interface/lib/src/platform_interface/platform_interface_messaging.dart:309]:
```suggestion
  /// Enables or disables Firebase Cloud Messaging message delivery metrics export to BigQuery.
  ///
  /// On iOS, you need to follow [this guide](https://firebase.google.com/docs/cloud-messaging/understand-delivery?platform=ios#enable_delivery_data_export_for_background_notifications)
  /// in order to export metrics to BigQuery.
  /// On Web, you need to setup a [service worker](https://firebase.google.com/docs/cloud-messaging/js/client) and call `experimentalSetDeliveryMetricsExportedToBigQueryEnabled(messaging, true)`
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]

File: packages/firebase_messaging/firebase_messaging_platform_interface/lib/src/platform_interface/platform_interface_messaging.dart (hunk 1 above)
Comment (Lyokone) [reply packages/firebase_messaging/firebase_messaging_platform_interface/lib/src/platform_interface/platform_interface_messaging.dart:309]: Fixed :)

File: docs/cloud-messaging/receive.md [hunk 2]
```diff
@@ -339,3 +339,64 @@ Here's how to use the second method:
...
+For Web, you need to change your service worker in order to use the v9 version of the SDK.
+The v9 version needs to be bundled, so you need to use a bundler like `esbuild` for instance 
+to get the service worker to work.
+See [the example app](https://github.com/firebase/flutterfire/blob/master/packages/firebase_messaging/firebase_messaging/example/bundled-service-worker) to see how to achieve this.
+
+Once you've migrated to the v9 SDK, you can use the following code:
+
+``` typescript
+const messaging = getMessaging(app);
+experimentalSetDeliveryMetricsExportedToBigQueryEnabled(messaging, true);
```
Comment (Ehesp) [inline docs/cloud-messaging/receive.md:399]: We should probably put the import at the top here so it's explicit where this comes from.

Comment (Lyokone) [reply docs/cloud-messaging/receive.md:399]: Agree, I'm adding that

**Review by Salakar — APPROVED** [review]

**Pull request merged**